database/sqlite3/sqlite3_test.go

```diff
@@ -160,3 +160,23 @@ func TestNoTxWrapInvalidValue(t *testing.T) {
 		assert.Contains(t, err.Error(), "invalid syntax")
 	}
 }
+
+func TestMigrateWithDirectoryNameContainsWhitespaces(t *testing.T) {
+	dir, err := ioutil.TempDir("", "directory name contains whitespaces")
+	if err != nil {
+		return
+	}
+	defer func() {
+		if err := os.RemoveAll(dir); err != nil {
+			t.Error(err)
+		}
+	}()
+	t.Logf("DB path : %s\n", filepath.Join(dir, "sqlite3.db"))
+	p := &Sqlite{}
+	addr := fmt.Sprintf("sqlite3://%s%s", "file:", filepath.Join(dir, "sqlite3.db"))
```

Comment:
According to SQLite [document](https://www.sqlite.org/uri.html):
> SQLite uses the "file:" URI syntax to identify database files. SQLite strives to interpret file: URIs in exactly the same way as popular web-browsers



I found that by prefixing `file:` to the DSN, then SQLite driver will interpret URI correctly. So unescaping the connection string is no longer needed. This test could also be treated as a reference on how to handle directory name with whitespaces.

Reply:
Can you clean up this a bit? e.g.
```golang
dbPath := filepath.Join(dir, "sqlite3.db")
t.Log("DB path:", dbPath)
addr := fmt.Sprintf("sqlite3://file:%s", dbPath)
```

Reply:
Thanks for the suggestion! Already modified the code.